src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java

```diff
@@ -92,8 +93,15 @@ private StrimziKafkaContainer(CompletableFuture<String> imageName) {
 
     @Override
     protected void doStart() {
-        if (!imageNameProvider.isDone()) {
-            imageNameProvider.complete(KafkaVersionService.strimziTestContainerImageName(kafkaVersion));
+        if (!this.imageNameProvider.isDone()) {
+            this.imageNameProvider.complete(KafkaVersionService.strimziTestContainerImageName(this.kafkaVersion));
+        }
```

Comment:
Why is this first? I would expect the check which does nothing and just throws exception to be first before calling some other methods. Or is it this method which sets the fields used below?

Reply:
>  Or is it this method which sets the fields used below?

Yes :)